doc/api/assert.md

```diff
@@ -147,6 +147,125 @@ try {
 }
 ```
 
+## Class: assert.CallTracker
+
+### `new assert.CallTracker([options])`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `options` {string|Object}
+  * `onExit` {boolean} **Default** `false`
+
+Creates a new [`CallTracker`][] object which can be used to track if functions
+were called a specific number of times. The `tracker.verify()` must be called
+for the verification to take place. The usual pattern would be to call it in the
+`process` `exit` handler.
+
+`options` can include an `onExit` value that, when `true` attaches
+`tracker.verify()` to the `process` `exit` handler.
+
+```js
+const assert = require('assert');
+
+const tracker = new assert.CallTracker({ onExit: true });
+
+function func() {}
+
+// callsfunc() must be called exactly 1 time before tracker.verify()
+const callsfunc = tracker.calls(func, 1);
+
+callsfunc();
+
+// Calls tracker.verify() and verifies if all tracker.calls() functions have
+// been called exact times.
+process.exit();
+```
+
+### `assert.CallTracker.calls([fn][, exact])`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `fn` {Function} **Default** A no-op function.
+* `exact` {number} **Default** `1`.
+* Returns: {Function} that wraps `fn`.
+
+The wrapper function is expected to be called exactly `exact` times. If the
+function is not called exactly `exact` times before
+[`assert.CallTracker.verify()`][] is called, an exception is thrown.
+
+```js
+const assert = require('assert');
+
+// Creates call tracker.
+const tracker = assert.createCallTracker();
+
+function func() {}
+
+// Returns a function that wraps func() that must be called exact times
+// before tracker.verify().
+const callsfunc = tracker.calls(func);
+```
+
+### `assert.CallTracker.report()`
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {Array} of objects containing information about the wrapper functions
+returned by [`assert.CallTracker.calls()`][].
+
```

Comment:
Would be good to document the object properties here and show them in the example